### PR TITLE
Theo/create page

### PR DESCRIPTION
**Pull request by theodorebugnet**
Create page draft. The form is mostly there, but a) link generation method in utils is a placeholder and b) a component needs to be added abstracting away the submission flow (including displaying the link once the "contract" has been "created").

Done:
 - Partial UI for Create
 - i18n
 - Price updating component
 - Metamask connection

Notable TODOs:
 - Radio buttons aren't working properly.
 - I didn't find any good premade switches in bootstrap or react-bootstrap. So it's a normal radio button for now, might need to either restyle these or find a good library. (Same component will probably be used on the positions page, to toggle between "Sold only"/"All"/"Bought only" and active/historical).
 - Centering/sizing is off, not sure what the cleanest way to achieve this is with bootstrap and modern CSS.

### CODE REVIEW

File: src/pages/create.tsx [hunk 1]
```diff
@@ -1,11 +1,228 @@
...
+    None,
+    Processing,
+    Success,
+    Failure,
+}
+
+const Create = ({ t }: { readonly t: TFunction }): ReactElement => {
+    const isConnected = useSelector((state: AppState) => state.user.isConnected);
+
+    const defaultState: Option = {
```
Comment (danhper) [inline src/pages/create.tsx]: I feel like keeping this sort of thing out of the view is cleaner.
For `User` I did this: https://github.com/interlay/xoptsv3-ui/blob/dev/src/lib/entities/user.ts

The idea is to keep the `types` for type only things (e.g. `Optional`) and have what would be the model in a typical MVC in `entities`.
Let me know what you think.

File: .eslintrc.json [hunk 2]
```diff
@@ -48,7 +48,7 @@
           "max-len": [
             "error",
             {
-              "code": 120
+              "code": 100
```
Comment (danhper) [inline .eslintrc.json:53]: I am fine with either but if we change to 100 we should also change prettier config in package.json.

File: src/common/utils.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Dispatch } from "react";
+import { updateBTCSpotPriceAction } from "../lib/actions";
+import { PriceActions } from "../lib/types/index";
+import { Option } from "../lib/types/option";
+
+export const updateBTCSpotPrice = async (dispatch: Dispatch<PriceActions>): Promise<void> => {
+    const mockRandom = Math.random() * 100 - 50;
+    const newPrice: number = await new Promise((resolve) => setTimeout(() => resolve(10500 + mockRandom), 500));
```
Comment (danhper) [inline src/common/utils.ts]: We might need to double this `10500` now :joy:

File: src/lib/types/actions/prices.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export const UPDATE_BTC_SPOT = "UPDATE_BTC_SPOT";
+
+export interface UpdateBTCSpot {
```
Comment (danhper) [inline src/lib/types/actions/prices.ts:3]: That's totally on me but thinking a bit more about it I think I would rather keep the action type and the action itself grouped in `lib/actions`.
What do you think?

**Review by danhper — APPROVED** [review]
I think it looks good. I left a few minor comments but I think we can fix that later too!

File: src/pages/create.tsx (hunk 1 above)
Comment (theodorebugnet) [reply src/pages/create.tsx]: Definitely good call, I didn't think of it while building but will update in the morning

File: src/lib/types/actions/prices.ts (hunk 4 above)
Comment (theodorebugnet) [reply src/lib/types/actions/prices.ts:3]: I'm not sure what the best practice around here is. Either way there are a lot of files to keep open whenever adding actions or action types, so personally I don't mind either way.
I'm pretty certain you have a lot more experience with react/redux than me so I'd say your call here on what will be more convenient long term. I'm happy to update my PR if you prefer to change it.

File: src/common/utils.ts (hunk 3 above)
Comment (theodorebugnet) [reply src/common/utils.ts]: I'm gonna say it helps make it clear that it's a mock, yes, that's definitely why I used that.

**Pull request merged**